Crypto/Random/Entropy/Windows.hs

```diff
@@ -43,26 +35,52 @@ instance EntropySource WinCryptoAPI where
     entropyGather WinCryptoAPI ptr n = do
         mctx <- cryptAcquireCtx
         case mctx of
-            Nothing  -> error "cannot re-grab win crypto api"
+            Nothing  -> do
+                lastError <- getLastError
+                fail $ "cannot re-grab win crypto api: error " ++ show lastError
```

Comment:
This used to use `error`, but I changed it to `fail` because `cryptAcquireCtx` was using `fail`.